lnrpc/signrpc/signer_server.go

```diff
@@ -383,3 +400,75 @@ func (s *Server) ComputeInputScript(ctx context.Context,
 
 	return resp, nil
 }
+
+// SignMessage signs a message with the key specified in the key locator. The
+// returned signature string is `zbase32` encoded and pubkey recoverable,
+// meaning that only the message digest and signature are needed for
+// verification.
+func (s *Server) SignMessage(ctx context.Context,
+	in *SignMessageReq) (*SignMessageResp, error) {
+
+	if in.Msg == nil {
+		return nil, fmt.Errorf("a message to sign MUST be passed in")
+	}
+
+	if in.KeyLoc == nil {
+		return nil, fmt.Errorf("a key locator MUST be passed in")
+	}
+
+	// Derive the private key we'll be using for signing.
+	keyLocator := keychain.KeyLocator{
+		Family: keychain.KeyFamily(in.KeyLoc.KeyFamily),
+		Index:  uint32(in.KeyLoc.KeyIndex),
+	}
+	privKey, err := s.cfg.KeyRing.DerivePrivKey(keychain.KeyDescriptor{
+		KeyLocator: keyLocator,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("can't derive private key: %v", err)
+	}
+
+	// The signature is over the double-sha256 hash of the prefixed message.
+	in.Msg = append(signedMsgPrefix, in.Msg...)
+	digest := chainhash.DoubleHashB(in.Msg)
+
+	// Create the pubkey-recoverable signature using the compressed pubkey.
```

Comment:
I think for this, we just want a raw regular signature: so no compact encoding, and no `zbase32`, just a raw ECDSA signature.